ebt/Dialog/BtSender.vala

```diff
@@ -0,0 +1,351 @@
+            model.get (iter, 0, out filename);
+                if (filename == file) {
+                    exist = true;
+                }
+                return false;
+            });
+            if (exist) {
+                return;
+            }
+            Gtk.TreeIter iter;
+            liststore.append (out iter);
+            liststore.set (iter, 0, file);
+            total_n_current (true);
+        }
+    }
+    private bool next_file () {
+        Gtk.TreeIter iter;
+        if (liststore.get_iter_from_string (out iter, current_file.to_string ())) {
+            liststore.get (iter, 0, out file_path);
+            send_file.begin ();
+            total_n_current ();
+            return true;
+        }
+        return false;
+    }
+    private void total_n_current (bool total = false) {
+        total_file = 0;
+        int current = 0;
+        liststore.foreach ((model, path, iter) => {
+            File file;
+            model.get (iter, 0, out file);
+            if (file == file_path) {
+                current = total_file;
+            }
+            total_file++;
+            return false;
+        });
+        if (!total) {
+            current_file = current + 1;
+        }
+    }
+
+    private async void create_season () {
+        try {
+            connection = yield GLib.Bus.get (BusType.SESSION);
+            client_proxy = yield new GLib.DBusProxy (connection, GLib.DBusProxyFlags.DO_NOT_LOAD_PROPERTIES | GLib.DBusProxyFlags.DO_NOT_CONNECT_SIGNALS, null, "org.bluez.obex", "/org/bluez/obex", "org.bluez.obex.Client1");
+            VariantBuilder builder = new VariantBuilder (VariantType.DICTIONARY);
+            builder.add ("{sv}", "Target", new Variant.string ("opp"));
+            Variant parameters = new Variant ("(sa{sv})", device.address, builder);
+            Variant variant_client = yield client_proxy.call ("CreateSession", parameters, GLib.DBusCallFlags.NONE, -1);
+            variant_client.get ("(o)", out s_session);
+            session = yield new GLib.DBusProxy (connection, GLib.DBusProxyFlags.NONE, null, "org.bluez.obex", s_session, "org.bluez.obex.ObjectPush1");
+            send_file.begin ();
+        } catch (Error e) {
+            GLib.warning (e.message);
+        }
+    }
+    private async void remove_session () {
+        try {
+            yield client_proxy.call ("RemoveSession", new Variant ("(o)", s_session), GLib.DBusCallFlags.NONE, -1);
+        } catch (Error e) {
+            GLib.warning (e.message);
+        }
+    }
+    private async void send_file () {
+        path_label.set_markup (_("<b>From</b>: %s").printf (file_path.get_parent ().get_path ()));
+        device_label.set_markup (_("<b>To</b>: %s").printf (GLib.Markup.escape_text (device.name)));
+        icon_label.set_from_gicon (new ThemedIcon (device.icon == null? "bluetooth" : device.icon), Gtk.IconSize.LARGE_TOOLBAR);
+        progress_label.label = _("Sending… (%i/%i)").printf (current_file, total_file);
```

Comment:
Inconsistent with previous in Device.vala:
```
_("Sending… %s of %s").printf (format_size (transfer.transferred), format_size (transfer.size));
```
Do we need two functions for this?

Reply:
More info details for dialog